utils/pubsub/pubsubClient.ts

```diff
@@ -0,0 +1,17 @@
+import { PubSub } from '@google-cloud/pubsub';
+
+// Set up authentication and initialize PubSub client
+const pubsub = new PubSub({ projectId: process.env.PROJECT_ID });
+
+
+export function publishMessage(topicName: string, data: any, msgtype: string) {
```

Comment:
Are there absolutely no constraints on the type of `data`?

(LOGAF: 4/10)

Reply:
I can set it to be dictionary. Anyone who has to send a pubsub message will have to give a key -value pair, which I think is good

Reply:
See `types/AuthInfo.d.ts` for reference to do that.
And yes, please do that.
And if this is a function that can be called in many different contexts, then I would suggest that you also add a function description comment on top of it:

```suggestion
/**
* publishMessage publishes a message to our GCP PubSub queue yadayada
* @param topicName {string} is blah blah
* @param data {myDataType} is blah blah blah
* @param msgType {string} is for yadiyada
*/
export function publishMessage(topicName: string, data: any, msgType: string) {
```